workers/unity/Packages/io.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Partials/Improbable.Gdk.TransformSynchronization.CompressedQuaternion

```diff
@@ -0,0 +1,215 @@
+private const float SqrtHalf = 0.70710678118f;
+
+public static readonly CompressedQuaternion Identity = FromUnityQuaternion(Quaternion.identity);
+
+public static bool operator ==(CompressedQuaternion a, CompressedQuaternion b) => a.Equals(b);
+public static bool operator !=(CompressedQuaternion a, CompressedQuaternion b) => !a.Equals(b);
+
+public override bool Equals(object obj)
+{
+    if (ReferenceEquals(null, obj))
+    {
+        return false;
+    }
+
+    return obj is CompressedQuaternion other && Equals(other);
+}
+
+public bool Equals(CompressedQuaternion other)
+{
+    return Data == other.Data;
+}
+
+public override int GetHashCode()
+{
+    return Data.GetHashCode();
+}
+
+/// <summary>
+///     Returns the string representation of the CompressedQuaternion.
+/// </summary>
+public override string ToString()
+{
+    return $"CompressedQuaternion{ToUnityQuaternion().ToString()}";
+}
+
+/// <summary>
+///     Decompresses a quaternion from a packed uint32 to an unpacked Unity Quaternion.
+/// </summary>
+/// <param name="compressedQuaternion">
+///     The CompressedQuaternion to decompress.
+/// </param>
+/// <remarks>
+///     <para>
+///         The underlying uint contains the "smallest three" components of a quaternion. The first two bits of
+///         of the uint encode which component is the largest value. This component's value is then computed as
+///         1 minus the sum of the squares of the smallest three components.
+///     </para>
+///     <para>
+///         This method is marked as unsafe due to the use of stack allocation.
+///     </para>
+/// </remarks>
+/// <returns>
+///     A decompressed Unity Quaternion.
+/// </returns>
+public UnityEngine.Quaternion ToUnityQuaternion()
+{
+    // The raw uint representing a compressed quaternion.
+    var compressedValue = Data;
+
+    // Borrow array from pool.
+    // q[x, y, z, w]
+    var q = QuaternionPool.Borrow();
```

Comment:
Why have we gone with a pool here? Over the simple stackalloc or Unity's NativeArray?

Reply:
stackalloc is unsafe, so can't be used in generated code

Reply:
removed `QuaternionPool` in favour of `float4`